crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,181 @@
+    ExecutableNotFoundInPath {
+        /// High-level error message meant to be printed to the user (install tips etc).
+        message: String,
+
+        /// Name used for the executable search.
+        executable_name: String,
+
+        /// Value of the `PATH` environment variable, if any.
+        search_path: String,
+    },
+
+    /// Failed to find Rerun Viewer executable at explicit path.
+    #[error("Failed to find Rerun Viewer executable at {executable_path:?}")]
+    ExecutableNotFound {
+        /// Explicit path of the executable (specified by the caller).
+        executable_path: String,
+    },
+
+    /// Other I/O error.
+    #[error("Failed to spawn the Rerun Viewer process: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+impl std::fmt::Debug for SpawnError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Due to how recording streams are initialized in practice, most of the time `SpawnError`s
+        // will bubble all the way up to `main` and crash the program, which will call into the
+        // `Debug` implementation.
+        //
+        // Spawn errors include a user guide, and so we need them to render in a nice way.
+        // Hence we redirect the debug impl to the display impl generated by `thiserror`.
+        <Self as std::fmt::Display>::fmt(self, f)
+    }
+}
+
+/// Spawns a new Rerun Viewer process ready to listen for TCP connections.
+///
+/// Refer to [`SpawnOptions`]'s documentation for configuration options.
+///
+/// This only starts a Viewer process: if you'd like to connect to it and start sending data, refer
+/// to [`crate::RecordingStream::connect`] or use [`crate::RecordingStream::spawn`] directly.
+pub fn spawn(opts: &SpawnOptions) -> Result<(), SpawnError> {
+    use std::{net::TcpStream, process::Command, time::Duration};
+
+    // NOTE: It's indented on purpose, it just looks better and reads easier.
+    const EXECUTABLE_NOT_FOUND: &str = //
+    "
+    You can install binary releases of the Rerun Viewer:
+    * Using `cargo`: `cargo binstall rerun-cli` (see https://github.com/cargo-bins/cargo-binstall)
+    * Via direct download from our release assets: https://github.com/rerun-io/rerun/releases/latest/
+    * Using `pip`: `pip3 install rerun-sdk` (warning: pip version has slower start times!)
+
+    For more information, refer to our complete install documentation over at:
+    https://rerun.io/docs/getting-started/installing-viewer
+    ";
+
+    let port = opts.port;
+    let connect_addr = opts.connect_addr();
+    let memory_limit = &opts.memory_limit;
+    let executable_path = opts.executable_path();
+
+    if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+        re_log::info!(
+            addr = %opts.listen_addr(),
+            "A process is already listening at this address. Assuming it's a Rerun Viewer."
+        );
+        return Ok(());
+    }
```

Comment:
To return `Ok` when perhaps there is a completely unrelated (non-rerun) process already occupying the port seems like trouble waiting to happen…

To me it makes sense it it is a special error (`SpawnError::AddressAlreadyInUse`) and have the caller do the assuming instead, but it's not a blocker. At least the logging is loud and clear!

Reply:
We've used the same trick in the Python SDK for months without issue.
I'll make an issue to introduce a proper Rerun handshake, but this is out of the scope of this PR.